crucible-llvm-syntax/src/Lang/Crucible/LLVM/Syntax.hs

```diff
@@ -0,0 +1,182 @@
+        loc <- Parse.position
+        Parse.depCons Parse.posNat $ \(Parse.BoundedNat w) -> do
+          assign <- Parse.operands (Ctx.Empty Ctx.:> BoolRepr Ctx.:> LLVMPointerRepr w Ctx.:> LLVMPointerRepr w)
+          let (rest, p2) = Ctx.decompose assign
+          let (rest', p1) = Ctx.decompose rest
+          let (Ctx.Empty, b) = Ctx.decompose rest'
+          let expr = Ext.LLVM_PointerIte w b p1 p2
+          Some <$> Parse.freshAtom loc (Reg.EvalApp (Expr.ExtensionApp expr))
+
+      Atom.AtomName "alloca" -> do
+        loc <- Parse.position
+        (align, assign) <- 
+          Parse.cons
+            parseAlign
+            (Parse.operands (Ctx.Empty Ctx.:> BVRepr ?ptrWidth))
+        let (Ctx.Empty, sz) = Ctx.decompose assign
+        let stmt = Ext.LLVM_Alloca ?ptrWidth mvar sz align (show (pretty loc))
+        Some <$> Parse.freshAtom loc (Reg.EvalExt stmt)
+
+      Atom.AtomName "load" -> Parse.describe "LLVM load arguments" $ do
+        loc <- Parse.position
+        (align, (memTy, assign)) <- 
+          Parse.cons
+            parseAlign
+            (Parse.cons
+              parseMemType
+              (Parse.operands (Ctx.Empty Ctx.:> LLVMPointerRepr ?ptrWidth)))
+        llvmTypeAsRepr memTy $ \tyRep -> do
+          case Mem.toStorableType memTy of
+            Nothing -> empty
+            Just storTy -> do
+              let (Ctx.Empty, ptr) = Ctx.decompose assign
+              let stmt = Ext.LLVM_Load mvar ptr tyRep storTy align
+              Some <$> Parse.freshAtom loc (Reg.EvalExt stmt)
+
+      Atom.AtomName "store" -> Parse.describe "LLVM store arguments" $ do
+        loc <- Parse.position
+        Parse.depCons parseAlign $ \align ->
+          Parse.depCons parseMemType $ \memTy -> do
+            llvmTypeAsRepr memTy $ \tyRep -> do
+              assign <- Parse.operands (Ctx.Empty Ctx.:> LLVMPointerRepr ?ptrWidth Ctx.:> tyRep)
+              case Mem.toStorableType memTy of
+                Nothing -> empty
+                Just storTy -> do
+                  let (rest, val) = Ctx.decompose assign
+                  let (Ctx.Empty, ptr) = Ctx.decompose rest
+                  let stmt = Ext.LLVM_Store mvar ptr tyRep storTy align val
+                  Some <$> Parse.freshAtom loc (Reg.EvalExt stmt)
+
+      _ -> empty
+  where
+    parseAlign :: MonadSyntax Atomic m => m Alignment
+    parseAlign = Parse.describe "alignment" $ do
+      s <- Parse.atomName
+      unless (s == Atom.AtomName "none") Parse.cut
+      pure DataLayout.noAlignment
+
+    parseMemType :: MonadSyntax Atomic m => m MemType
+    parseMemType = Parse.describe "LLVM type" $ do
+      nm <- Parse.atomName
+      case nm of
+        Atom.AtomName "i8" -> pure (MemType.IntType 8)
+        Atom.AtomName "i16" -> pure (MemType.IntType 16)
+        Atom.AtomName "i32" -> pure (MemType.IntType 32)
+        Atom.AtomName "i64" -> pure (MemType.IntType 64)
+        Atom.AtomName "ptr" -> pure MemType.PtrOpaqueType
```

Comment:
It would be worth documenting the list of atoms that represent types in the `README` as well.